### PR TITLE
TNS watcher

### PR DESCRIPTION
**Pull request by dmitryduev**
This PR implements a sentinel service that constantly monitors [TNS](https://wis-tns.weizmann.ac.il/search?format=csv&num_page=50) and saves/updates the most recent entries in the `kowalski.TNS` collection.

### CODE REVIEW

**Review by dannygoldstein — APPROVED** [review]

**Review by acrellin — APPROVED** [review]
This looks good. How do the TNS queries work? Are the most recent additions always in the first pages?

**Comment by dmitryduev** [comment]
@acrellin yes, page=0 is the most recent stuff

File: kowalski/tns_watcher.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    if grab_all:
+        # grab the latest data:
+        url = f'https://wis-tns.weizmann.ac.il/search?format=csv&num_page=50&page=0'
+        data = pd.read_csv(url)
+        # TNS outputs a maximum of 50 entries per call/page
+        num_pages = data["ID"].max() // 50
+
+    for num_page in range(num_pages):
+        log(f"Digesting page #{num_page+1} of {num_pages}...")
```
Comment (stefanv) [inline kowalski/tns_watcher.py:122]: Seems to be making one more call (to page 0) than is necessary.

File: kowalski/tns_watcher.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        ],
+        background=True
+    )
+
+    log("Fetching data...")
+
+    # grab the last 20 pages (with 50 entries each) by default
+    num_pages = 20
+
+    if grab_all:
```
Comment (stefanv) [inline kowalski/tns_watcher.py:114]: This is a minor point, but I would restructure this a bit for clarity:

```
if grab_all:
    num_pages = get_TNS_pages()
else:
    num_pages = 20
```

**Review by stefanv — APPROVED** [review]

File: kowalski/tns_watcher.py (hunk 1 above)
Comment (dmitryduev) [reply kowalski/tns_watcher.py:122]: Changed to request the 5 latest entries, which is hopefully faster

File: kowalski/tns_watcher.py (hunk 2 above)
Comment (dmitryduev) [reply kowalski/tns_watcher.py:114]: Fixed

**Pull request merged**